docs/v3/how-to-guides/deployments/prefect-yaml.mdx

```diff
@@ -518,6 +518,16 @@ The above command deploys:
 `my-deployment`
 - all deployments starting with `dep` and ending in `prod`.
 
+### Non-interactive deployment
+
+For CI/CD pipelines and automated environments, use the `--no-prompt` flag to skip interactive prompts:
+
+```bash
+prefect --no-prompt deploy --name my-deployment
+```
+
+This prevents the command from hanging on prompts and will fail clearly if required information is missing.
+
```

Comment:
i was considering a dedicated how-to since this is a top level CLI feature but i wasn't sure what else would go in it

so we can always move this there and link out if we find more content for a CLI-specific How To